### PR TITLE
Fix concurrency issue with gateway RPM plugin

### PR DESCRIPTION
**Pull request by varungup90**
https://github.com/aibrix/aibrix/issues/254

### CODE REVIEW

File: pkg/cache/cache.go [hunk 1]
```diff
@@ -159,9 +159,11 @@ func (c *Cache) updatePod(oldObj interface{}, newObj interface{}) {
 		return
 	}
 
+	delete(c.pods, oldPod.Name)
+	c.pods[newPod.Name] = newPod
 	c.deletePodAndModelMapping(oldPod.Name, oldModelName)
 	c.addPodAndModelMapping(newPod.Name, newModelName)
-	klog.Infof("POD UPDATED. %s/%s %s", oldPod.Namespace, oldPod.Name, newPod.Status.Phase)
```
Comment (Jeffwan) [inline pkg/cache/cache.go:164]: can we change this to v(4). We can use log level to control the messages. I see lots of logs in the controller manager

File: pkg/plugins/gateway/gateway.go [hunk 2]
```diff
@@ -182,6 +183,17 @@ func (s *Server) HandleRequestHeaders(ctx context.Context, req *extProcPb.Proces
 			fmt.Sprintf("pre query: error on checking rpm: %v", err.Error())), username, targetPodIP
 	}
 
+	rpm, code, err := s.incrRPM(ctx, username)
```
Comment (Jeffwan) [inline pkg/plugins/gateway/gateway.go:186]: What's the purpose here?

File: pkg/cache/cache.go (hunk 1 above)
Comment (varungup90) [reply pkg/cache/cache.go:164]: Done.

File: pkg/plugins/gateway/gateway.go (hunk 2 above)
Comment (varungup90) [reply pkg/plugins/gateway/gateway.go:186]: Previously increment was done at the end of the request, so problem was that with burst of request. Now rpm is tracked as the request arrives.

**Pull request merged**